vignettes/qgisprocess.Rmd

```diff
@@ -10,73 +10,92 @@ output:
+
 To facilitate using **qgisprocess**, we have created a docker image that already comes with the needed software packages.
 You can pull it from dockerhub by running:
 
 ```{sh, eval=FALSE}
 docker pull geocompr/geocompr:qgis-ext
 ```
 
-For a more detailed introduction on how to use docker with **qgisprocess** have a look a the corresponding vignette.
+For a more detailed introduction on how to use docker with **qgisprocess**, have a look a the corresponding vignette.
 
-# Basic usage
+# Package configuration
 
-Since **qgisprocess** depends on the command line tool `qgis_process`, it already tries to detect `qgis_process` on your system when it is being attached, and complains if it cannot find it.
+Since **qgisprocess** depends on the command line tool `qgis_process`, it already tries to detect `qgis_process` on your system when it is being loaded, and complains if it cannot find it.
 
 ```{r}
 library("qgisprocess")
 ```
 
-If you want to find out, how **qgisprocess**  detected `qgis_process`, run `qgis_configure()`.
+More specifically, **qgisprocess** tries to make use of the previously saved state of the package, including the filepath of `qgis_process`, the QGIS version and the list of known algorithms.
+Both the filepath and the QGIS version are checked, as well as other conditions.
+
+So the **main point** is that usually you don't have to do any configuration yourself, unless there's a message telling you to do so.
+
+If you want to find out more details about this process, e.g. how **qgisprocess**  detected `qgis_process`, run `qgis_configure(use_cached_data = TRUE)`.
 
 ```{r config}
-qgis_configure()
+qgis_configure(use_cached_data = TRUE)
 ```
 
-Here, `qgis_process` was detected in the PATH variable.
-However, `qgis_configure()` first tries if the R option `qgisprocess.path` or the global environment variable `R_QGISPROCESS_PATH` has been set.
-This already indicates that you can specify one of these variables in case `qgis_process` has not been installed in one of the most common locations or if there are multiple QGIS versions available (both of which might sometimes happen under Windows).
-If this is the case, set `options(qgisprocess.path = '/path/to/qgis_process')` and re-run `qgis_configure()`.
+If needed the cache will be rebuilt upon loading the package.
+Example reasons for automated rebuild include changed QGIS version, changed `qgis_process` location, a non-matching user-setting (e.g. the option `qgisprocess.path`) and a changed state of processing provider plugins.
+
+Rebuilding the cache can be triggered manually by running `qgis_configure()` (its default is `use_cached_data = FALSE`).
+
+To determine the location of `qgis_process`, `qgis_configure()` first tries if the R option `qgisprocess.path` or the global environment variable `R_QGISPROCESS_PATH` has been set.
```

Comment:
Hi @jannes-m the approach is robust: if the _option_ is set, it takes precedence. Only if it is not set, the environment variable is checked. Maybe you want to say that in the vignette indeed. You can see the flow in the code of `qgis_query_path()`.

Also, the path set in one of those two ways must be an effectively working `qgis_process` instance (it is called as a test), else it will not be accepted and directly result in an error.

Reply:
> directly result in an error

EDIT: it will print the error, and proceed to the next step in the hope of being successful. So if `qgisprocess.path` fails because there it sets a non-working path, `qgis_query_path()` will proceed to trying `R_QGISPROCESS_PATH`.